charts/zeebe-cluster-helm/test/integration_test.go

```diff
@@ -0,0 +1,126 @@
+// +build integration
+
+package test
+
+import (
+	"context"
+	"fmt"
+	"github.com/gruntwork-io/terratest/modules/logger"
+	"github.com/gruntwork-io/terratest/modules/retry"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"google.golang.org/grpc"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/camunda-cloud/zeebe/clients/go/pkg/zbc"
+	"github.com/gruntwork-io/terratest/modules/helm"
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	"github.com/gruntwork-io/terratest/modules/random"
+)
+
+type integrationTest struct {
+	suite.Suite
+	chartPath   string
+	release     string
+	namespace   string
+	kubeOptions *k8s.KubectlOptions
+}
+
+func TestIntegration(t *testing.T) {
+	t.Parallel()
+
+	chartPath, err := filepath.Abs("../")
+	require.NoError(t, err)
+
+	namespace := "zeebe-" + strings.ToLower(random.UniqueId())
+
+	// currently enforce the test context to be the Zeebe team's cluster, but could be change to some CI cluster
+	kubeOptions := k8s.NewKubectlOptions("gke_zeebe-io_europe-west1-b_zeebe-cluster", "", namespace)
+
+	suite.Run(t, &integrationTest{
+		chartPath:   chartPath,
+		release:     "zeebe-cluster-helm-it",
+		namespace:   namespace,
+		kubeOptions: kubeOptions,
+	})
+}
+
+func (s *integrationTest) SetupTest() {
+	k8s.CreateNamespace(s.T(), s.kubeOptions, s.namespace)
+}
+
+func (s *integrationTest) TearDownTest() {
+	k8s.DeleteNamespace(s.T(), s.kubeOptions, s.namespace)
+}
+
+func (s *integrationTest) TestGatewayConnection() {
```

Comment:
:rocket: This test looks quite easy to do really cool!

Reply:
Alternative to that:

 * https://github.com/helm/chart-testing/blob/main/doc/ct_install.md with https://helm.sh/docs/topics/chart_tests/
 * Example: https://github.com/camunda-community-hub/camunda-helm/blob/main/.github/workflows/ci.yaml#L68-L81

Reply:
This means we could define the tests as helm tests, similar to this https://github.com/camunda-community-hub/camunda-cloud-helm/blob/main/charts/ccsm-helm/charts/zeebe/templates/tests/test-connection.yaml (maybe just requesting the topology etc. )

These kind of tests can be run by all consumers of the charts, it is built-in.

Reply:
I think the Go tests are more powerful in terms of what you can verify. The Helm tests require you to launch a container and run the verification in the container. Of course, you could always run some Go binary in the container instead of the classic bash script example. I think it might be worth exploring that idea a bit more. 

I don't like the Helm examples - starting a container and running a bash script is not my idea of maintainable tests :smile: However, running a container which contains our suite of tests written in whatever language may be an option. Might be harder to debug and observe though (though not impossible either). I think it might be worth doing a small spike to explore the pros and cons, depending on how much time we have.